### ISSUE TITLE
NVDA does not read a table caption when any nested element is focused

### ISSUE THREAD

**Issue by mpreyskurantov** [nvaccess/nvda#5716]
Use the following HTML table markup to reproduce the issue:

```
<table>
    <caption>Test table</caption>
    <thead>
        <tr>
            <th><a href="#">First header</a></th>
            <th>Second header</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>First cell</td>
            <td>Second cell</td>
        </tr>
        <tr>
            <td>Third cell</td>
            <td>Forth cell</td>
        </tr>
    </tbody>
</table>
```

When the nested <a> tag is focused, the NVDA reader does not pronounce the table caption.

NVDA reads this markup in the following manner:

_table with 3 rows and 2 columns
row 1 column 1
First header
link_

**Comment by jcsteh** [comment]
In browse mode, the caption is just text at the top of the table. We don't read it when a nested element is focused in the same way that we don't read a heading at the top of a form when a form control gets focused. Doing otherwise would result in double speaking when entering the table. Also, captions can be quite long, so this would create verbosity issues for users.

**Comment by mpreyskurantov** [comment]
We suggest reading a caption at the very least when focusing the first nested focusable element for the first time. Otherwise, an end-user will not understand which table is now active/focused. For example, JAWS always reads it despite of the described problems.
